Scripts/script-StringReplace.yml

```diff
@@ -0,0 +1,82 @@
+commonfields:
+  id: StringReplace
+  version: -1
+name: StringReplace
+script: |-
+  if ((typeof args.data) !== 'string') {
+      return {
+          Type: entryTypes.error,
+          ContentsFormat: formats.text,
+          Contents: 'Invalid input - please make sure \'data\' is of type string'
+      }
+  }
+  var arr = [];
+  if (args.replaceAll === 'true') {
+      arr.push('g');
+  }
+  if (args.caseInsensitive === 'true') {
+      arr.push('i');
+  }
+  if (args.multiLine === 'true') {
+      arr.push('m');
+  }
+  var newValue = args.newValue === "''" ? "" : args.newValue;
+  var r = new RegExp(args.regex, arr.join(''));
+  if (r.test(args.data)) {
+      var val = args.data.replace(r, newValue);
+      setContext('StringReplace.Result', val);
+      return {
+          Type: entryTypes.note,
+          Contents: val,
+          ContentsFormat: formats.text
+      };
+  }
+  return {
+      Type: entryTypes.note,
+      Contents: 'No regex match to replace',
```

Comment:
return the original string. There can be cases where user just wants to replace some string, and if there is nothing to replace then return the original value

Reply:
And don't forget to setContext here too